### PR TITLE
fix: Refresh datasource structure on save

### LINKED ISSUE #17238: [Bug]: Explorer pane shows a datasource connection error even when error is rectified
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

**When** I have a datasource connectivity error, and I fix the error, **I don't expect to see the error** on the explorer pane after it has been rectified.
datasource-error

### Steps To Reproduce

1. Go to the datasources page and enter an invalid datasource configuration.
2. Hit "Test".
3. The connection test should fail and the connection error should be shown on the explorer pane on the left. You should see something like :
datasource-error
5. Now fix the connection error.
6. Although the connection error is fixed, you will still see the error on the explorer pane.
datasource-error

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ChandanBalajiBP**
## Description

> On saving the data source configuration for the first time, the client makes an API call(`/api/v1/datasources/*/structure*`) to fetch the schema of the datasource. But on subsequent saves API call to fetch schema were not triggered.

> Links - [Loom](https://www.loom.com/share/d71d351e3542483a9e6b247ebe506850)
> Schema when there is a configuration issue
image

> Schema when the data source is configured successfully. 
Screenshot 2022-11-22 at 3 34 35 PM

> TL;DR: This PR updates the schema of Datasource upon saving the configuration every time.

Fixes #17238 

Media
> Solution [Loom - ](https://www.loom.com/share/e40017c90b2844d58116eb5eca810f25)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by ayushpahwa — APPROVED** [review]

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks Good!
